src/babel/transformation/file/index.js

```diff
@@ -413,6 +413,30 @@ export default class File {
     return uid;
   }
 
+  addTemplateObject(helperName: string, strings: Array, raw: Array): Object {
+    // Generate a unique name based on the string literals so we dedupe
+    // identical strings used in the program.
+    var stringIds = raw.elements.map(function(string) {
+      return string.value;
+    });
+    var name = `${helperName}_${raw.elements.length}_${stringIds.join(",")}`;
```

Comment:
Yes. My assumption was this was a non-issue as the minimizer will optimize.
This of course assumes that you're using a minimizer... Perhaps we should
save a look up table on the file instance to shorter names?

On Tue, Jul 7, 2015 at 11:56 AM Sebastian McKenzie <notifications@github.com>
wrote:

> In src/babel/transformation/file/index.js
> <https://github.com/babel/babel/pull/1878#discussion_r34024402>:
>
> > @@ -413,6 +413,30 @@ export default class File {
> >      return uid;
> >    }
> >
> > +  addTemplateObject(helperName: string, strings: Array, raw: Array): Object {
> > +    // Generate a unique name based on the string literals so we dedupe
> > +    // identical strings used in the program.
> > +    var stringIds = raw.elements.map(function(string) {
> > +      return string.value;
> > +    });
> > +    var name = `${helperName}_${raw.elements.length}_${stringIds.join(",")}`;
>
> This will cause extremely long variable names for big strings.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/babel/babel/pull/1878/files#r34024402>.
>